InnerSource License: Added Robert Bosch GmbH as known instance

gruetter opened the pull request.

gruetter committed:
Added Robert Bosch GmbH as known instance

fioddor commented on patterns/2-structured/innersource-license.md:
```diff
@@ -66,6 +67,8 @@ The mentioned collaboration challenges include:
 
 It is worth mentioning that so far the software shared under this InnerSource license is mostly tooling, infrastructure, and tools lower in the stack.
 
+- **Robert Bosch GmbH** 
```
```suggestion
- **Deutsche Bahn AG** 
- **Robert Bosch GmbH** 
```

fioddor replied:
I suggest moving the list to the top of the section before the case details.

spier replied:
Good call @fioddor!

Formatting on this PR is not final yet, which is why we kept it as a "Draft".

Actually this PR happened as part of a (super rare) in-person conversation between @gruetter and me, so in a sense this is more of a mental note :)

spier replied:
I changed the formatting now, as suggested by @fioddor

spier committed:
Formatting Known Insstances

First listing all orgs in the Known Instances, then listing any additional details after that.

spier committed:
Fix linter issues

spier commented on patterns/2-structured/innersource-license.md:
```diff
@@ -32,6 +32,7 @@ At the time of sharing the source code, it can not be reliably predicted what th
 - Freedom over using the software leads to competition, and spread of ownership
 - There are legal contracts in place which cover the sharing of source code. These contracts are not standardized, so they create additional effort in negotiating and understanding for every project. The existing contracts may also not allow sharing source code in an open enough sense to support a true InnerSource approach.
 - Alternatively, there are no legal contracts in place but source code is shared informally. That might create uncertainty in cases where clarity about ownership and rights and obligations is needed.
+- **to be finalized**: wrong choice of license might inhibit transition to OSS
```
@gruetter do you ave time to finalize this description?

If not we can also merge the PR as is, so that we at least get the Known Instance added. You could then add further descriptions in a separate PR.

spier commented on patterns/2-structured/innersource-license.md:
```diff
@@ -51,6 +52,11 @@ The license simplifies the conversations within our organization about sharing s
+- **DB Systel**
+- **Robert Bosch GmbH**
+
+## DB Systel
```
I suspect we wanted to add a new section "## Robert Bosch GmbH" here as well?

gruetter replied:
FYI @spier : I have written a short paragraph and am currently clearing that internally.

spier replied:
@gruetter would you prefer to wait and add this to this PR, or shall we go ahead and merge, and handle future additions in a separate PR?

spier replied:
Opted to merge now. Will leave this thread open.

gruetter commented on patterns/2-structured/innersource-license.md:
```diff
@@ -32,6 +32,7 @@ At the time of sharing the source code, it can not be reliably predicted what th
 - Freedom over using the software leads to competition, and spread of ownership
 - There are legal contracts in place which cover the sharing of source code. These contracts are not standardized, so they create additional effort in negotiating and understanding for every project. The existing contracts may also not allow sharing source code in an open enough sense to support a true InnerSource approach.
 - Alternatively, there are no legal contracts in place but source code is shared informally. That might create uncertainty in cases where clarity about ownership and rights and obligations is needed.
+- **to be finalized**: wrong choice of license might inhibit transition to OSS
```
```suggestion
- Choosing a restrictive and/or copyleft license can constitute a barrier for InnerSource adoption. Specifically, limiting publication to the organisation might require a costly relicencing procedure prior to transitioning to Open Source.
```

spier replied:
Should be "relicensing" I think. The word license is a spelling issue for me every single time too. "Lizenz" would be so much easier :)

Also wondering if some elements of this argument should also be expressed in the "Resulting Context"?

From the definition of "Resulting Context":
> Often this section can include discussion of the next possible Patterns/problems introduced. 

Brainstorming something for the "Resulting Context" here:

> By applying this pattern, you should have a method in place that allows for simpler code sharing across legal entities within your organization.
> 
> If you envision a possible path from InnerSource to open source for the specific project in the future, these might be challenges that you need to consider: need for relicensing, ...

spier committed:
Merge branch 'main' into gruetter-patch-1

spier committed:
Reformatting pattern, to include Airbus reference (pulled in via last update from main)

spier committed:
Add another force related to possible relicensing when moved from InnerSource to open source

Co-authored-by: Georg Grütter <gruetter@gmail.com>

spier committed:
Spelling fix
